www/js/lib/zimArchive.js

```diff
@@ -142,31 +142,68 @@ define(['zimfile', 'zimDirEntry', 'util', 'utf8'],
      */
 
     /**
-     * Look for DirEntries with title starting with the given prefix.
+     * Look for DirEntries with title starting with the prefix of the current search object.
      * For now, ZIM titles are case sensitive.
      * So, as workaround, we try several variants of the prefix to find more results.
      * This should be enhanced when the ZIM format will be modified to store normalized titles
      * See https://phabricator.wikimedia.org/T108536
      * 
-     * @param {String} prefix
-     * @param {Integer} resultSize
-     * @param {callbackDirEntryList} callback
+     * @param {Object} search The current state.searches object
+     * @param {Integer} resultSize The number of dirEntries to find
+     * @param {callbackDirEntryList} callback The function to call with the result
+     * @param {Boolean} noInterim A flag to prevent callback until all results are ready (used in testing) 
      */
-    ZIMArchive.prototype.findDirEntriesWithPrefix = function(prefix, resultSize, callback) {
+    ZIMArchive.prototype.findDirEntriesWithPrefix = function (search, resultSize, callback, noInterim) {
+        var prefix = search.prefix;
         var that = this;
-        var prefixVariants = util.removeDuplicateStringsInSmallArray([prefix, util.ucFirstLetter(prefix), util.lcFirstLetter(prefix), util.ucEveryFirstLetter(prefix)]);
+        // Establish array of initial values that must be searched first
+        var startArray = [];
+        // Ensure a search is done on the string exactly as typed
+        startArray.push(prefix);
+        // Normalize any spacing and make string all lowercase
+        prefix = prefix.replace(/\s+/g, ' ').toLocaleLowerCase();
+        // Add lowercase string with initial uppercase (this is a very common pattern)
```

Comment:
Isn't this very common pattern already present in the `util.allCaseFirstLetters` results? (basic or full type)

Reply:
Indeed, but we want it to be searched first in the array of searches. It considerably speeds up the returning of results if the most common patterns are searched before the less likely (but still possible) combinations. The `removeDuplicateStringsInSmallArray` removes any duplicates that are lower down in the array.

Reply:
All right. This explanation could be added as a comment in the code.

Reply:
Will do.